tests/init.test.js

```diff
@@ -0,0 +1,135 @@
+const utils = require('./utils');
+const testRepo = require('./utils').testRepo;
+const fs = require('fs-extra');
+const path = require('path');
+const yaml = require('js-yaml');
+
+const { EXPLAIN_PLACEHOLDER } = require('../utils/common');
+
+// Tmp directories used in tests.
+let tmpDirs = Array();
+
+describe('tuture init', () => {
+
+  afterAll(() => {
+    tmpDirs.map(dir => fs.removeSync(dir));
+    process.chdir(path.join(__dirname, '..'));
+  });
+
+  // TODO: Test prompts when no args given.
+
+  describe('-y', () => {
+    const nonGitRepo = utils.createEmptyDir();
+    tmpDirs.push(nonGitRepo);
+    process.chdir(nonGitRepo);
+    const cp1 = utils.run(['init', '-y']);
+
+    it('should refuse to init outside a git repo', () => {
+      expect(cp1.status).toBe(1);
+    });
+
+    const gitRepo = utils.createGitRepo();
+    tmpDirs.push(gitRepo);
+    process.chdir(gitRepo);
+    const cp2 = utils.run(['init', '-y']);
+
+    it('should exit with status 0', () => {
+      expect(cp2.status).toBe(0);
+    });
+
+    it('should create .tuture/diff directory', () => {
+      const diffPath = path.join('.tuture', 'diff');
+      expect(fs.existsSync(diffPath)).toBe(true);
+      expect(fs.readdirSync(diffPath).length).toBe(testRepo.length);
+    });
+
+    it('should create correct tuture.yml with default values', () => {
+      expect(fs.existsSync('tuture.yml')).toBe(true);
+
+      const tuture = yaml.safeLoad(fs.readFileSync('tuture.yml'));
+      testTutureObject(tuture);
+    });
+  });
+
+  describe('--yes', () => {
+    const nonGitRepo = utils.createEmptyDir();
+    tmpDirs.push(nonGitRepo);
+    process.chdir(nonGitRepo);
+    const cp1 = utils.run(['init', '--yes']);
+
+    it('should refuse to init outside a git repo', () => {
+      expect(cp1.status).toBe(1);
+    });
+
+    const gitRepo = utils.createGitRepo();
+    tmpDirs.push(gitRepo);
+    process.chdir(gitRepo);
+    const cp2 = utils.run(['init', '--yes']);
+
+    it('should exit with status 0', () => {
+      expect(cp2.status).toBe(0);
+    });
+
+    it('should create .tuture/diff directory', () => {
+      const diffPath = path.join('.tuture', 'diff');
+      expect(fs.existsSync(diffPath)).toBe(true);
+      expect(fs.readdirSync(diffPath).length).toBe(testRepo.length);
+    });
+
+    it('should create correct tuture.yml with default values', () => {
+      expect(fs.existsSync('tuture.yml')).toBe(true);
+
+      const tuture = yaml.safeLoad(fs.readFileSync('tuture.yml'));
+      testTutureObject(tuture);
+    });
+  });
+});
+
+function testInit(args) {
```

Comment:
It seems  above two `describe` can use this `testInit` function to run test. 🌜

Reply:
Good catch! Actually I forgot to put `testInit` back to tests after debugging 🤯.